### PR TITLE
Mark/update settings to toggle auto detect schema changes notifications

### LINKED ISSUE #17830: Update settings to toggle auto-detect schema changes notifications
Update the connections settings page to enable/disable notification for auto-detect schema changes.

Design:
https://www.figma.com/file/liLQhcbpVHiEDW18kiXQe3/01_03_CONNECTIONS?node-id=2062%3A60213&t=EAjh7xtF8vK5jde6-1

Backend changes:
API: #18636
Impl: #18735

### PR DESCRIPTION
**Pull request by m19berger**
## What
Closes [#17830](https://github.com/airbytehq/airbyte/issues/17830)

## How
- Added new card to toggle auto-detect schema changes notifications

_For test purposes, you need to use the feature variable:_ REACT_APP_FEATURE_ALLOW_AUTO_DETECT_SCHEMA_CHANGES=true

## Loom 
https://www.loom.com/share/4fc94c68c2894538b6b73c19e2fcc48a

### CODE REVIEW

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab/ConnectionReplicationTab.module.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@use "scss/variables";
+@use "../../../../../scss/variables";
```
Comment (krishnaglick) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab/ConnectionReplicationTab.module.scss:1]: Why?

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/components/SchemaUpdateNotifications.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { Card } from "components/ui/Card";
+import { Switch } from "components/ui/Switch";
+
+import { useConnectionEditService } from "hooks/services/ConnectionEdit/ConnectionEditService";
+
+import styles from "./SchemaUpdateNotifications.module.scss";
+
+export const SchemaUpdateNotifications: React.FC = () => {
+  const { connection, updateConnection, connectionUpdating } = useConnectionEditService();
+  const [{ loading }, onChange] = useAsyncFn(async (checked: boolean) => {
```
Comment (krishnaglick) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/components/SchemaUpdateNotifications.tsx:15]: `connectionUpdating` handles this loading bit.

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
-@use "scss/colors";
-@use "scss/variables";
+@use "../../../../../../scss/colors";
+@use "../../../../../../scss/variables";
```
Comment (krishnaglick) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss:2]: Why this change?

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.tsx [hunk 4]
```diff
@@ -17,9 +17,9 @@ import { useCurrentWorkspace } from "hooks/services/useWorkspace";
 import { DbtCloudJob, isSameJob, useDbtIntegration, useAvailableDbtJobs } from "packages/cloud/services/dbtCloud";
 import { RoutePaths } from "pages/routePaths";
 
-import dbtLogo from "./dbt-bit_tm.svg";
+import dbtLogo from "../dbt-bit_tm.svg";
+import octaviaWorker from "../octavia-worker.png";
```
Comment (krishnaglick) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.tsx:21]: Why all these import sorts? Is this from our new rule?

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/ConnectionSettingsTab.tsx [hunk 5]
```diff
@@ -23,6 +24,7 @@ export const ConnectionSettingsTab: React.FC = () => {
 
   return (
     <div className={styles.container}>
+      <SchemaUpdateNotifications />
```
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/ConnectionSettingsTab.tsx:27]: this needs to only be rendered if the feature is turned on

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab/ConnectionReplicationTab.module.scss (hunk 1 above)
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab/ConnectionReplicationTab.module.scss:1]:
```suggestion
@use "scss/variables";
```

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionStatusTab/ConnectionStatusTab.module.scss [hunk 6]
```diff
@@ -1,5 +1,5 @@
-@use "../../../../scss/colors";
-@use "../../../../scss/variables";
+@use "../../../../../scss/colors";
```
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionStatusTab/ConnectionStatusTab.module.scss:1]:
```suggestion
@use "scss/colors";
```

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionStatusTab/ConnectionStatusTab.module.scss [hunk 7]
```diff
@@ -1,5 +1,5 @@
...
+@use "../../../../../scss/variables";
```
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionStatusTab/ConnectionStatusTab.module.scss:2]:
```suggestion
@use "scss/variables";
```

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss [hunk 8]
```diff
@@ -1,5 +1,5 @@
-@use "scss/colors";
-@use "scss/variables";
+@use "../../../../../../scss/colors";
```
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss:1]:
```suggestion
@use "scss/colors";
```

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss (hunk 3 above)
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss:2]:
```suggestion
@use "scss/variables";
```

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.tsx (hunk 4 above)
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.tsx:21]: why are we not using an SVG for the octaviaWorker image?

Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.tsx:21]: Why are we not using an SVG for the octaviaWorker image?

Curious why these are being imported this way and not as components?

**Review by teallarson — CHANGES_REQUESTED** [review]
On the right track!

Please remove the file rearrangements and save for a separate PR.  It makes the diff muddy and difficult to tell what has been changed in the code.

A few other notes:
- we should not be using relative imports for our root stylesheets
- this entire component needs to be hidden if the auto-detect schema changes feature is not turned on

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab/ConnectionReplicationTab.module.scss (hunk 1 above)
Comment (matter-q) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab/ConnectionReplicationTab.module.scss:1]: Sorry, it's replaced by my IDE

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss (hunk 3 above)
Comment (matter-q) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss:2]: Replaced by my IDE

Comment (matter-q) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.module.scss]: Same... IDE replacement :face_with_head_bandage:

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.tsx (hunk 4 above)
Comment (matter-q) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionTransformationTab/components/DbtCloudTransformationsCard.tsx]: I just changed the structure of ConnectionItemPage to make it more readable, but I will return to the previous implementation as discussed before

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/SchemaUpdateNotifications.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import { Switch } from "components/ui/Switch";
+
+import { useConnectionEditService } from "hooks/services/ConnectionEdit/ConnectionEditService";
+
+import styles from "./SchemaUpdateNotifications.module.scss";
+
+export const SchemaUpdateNotifications: React.FC = () => {
+  const { connection, updateConnection, connectionUpdating } = useConnectionEditService();
+  const [{ loading }, onChange] = useAsyncFn(async (checked: boolean) => {
+    await updateConnection({
```
Comment (YatsukBogdan1) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/SchemaUpdateNotifications.tsx:16]: @krishnaglick not sure about this one, but this just popped into my mind. do you think we should handle errors here? in case of failed request what will happen here?

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/ConnectionSettingsTab.tsx (hunk 5 above)
Comment (matter-q) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/ConnectionSettingsTab.tsx]: Fixed. Thanks!

**Review by YatsukBogdan1 — COMMENTED** [review]
left one question to @krishnaglick besides that LGTM. Will approve, after all other comments from guys will be fixed

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/components/SchemaUpdateNotifications.tsx (hunk 2 above)
Comment (matter-q) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/components/SchemaUpdateNotifications.tsx]: <Switch
        checked={connection.notifySchemaChanges}
        onChange={(event) => onChange(event.target.checked)}
        
        disabled={connectionUpdating} - should disable `switch` when the user interacts with `useConnectionEditService` 
        form different elements of UI ()
        
        loading={loading} - user will see a loading indicator to interact with the current toggle
      />
      
We can't use just one `connectionUpdating`. It's shared state from different components from this page (ConnectionSettingsTab)

Behavior with `connectionUpdating` for loading state:
loom.com/share/82276b66eef5496c83a1a52ed853370e
It looks buggy :face_with_head_bandage:

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/SchemaUpdateNotifications.tsx (hunk 9 above)
Comment (matter-q) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/SchemaUpdateNotifications.tsx]: Sounds good! I added toast notifications for the failed request

**Comment by krishnaglick** [comment]
After some investigation and a discussion with Mark:
1. Pull out the local loading state and use the loading state from the connection edit service.
2. Remove the error notification toast.
This will slim down the logic in the component to just the update call, and utilization of loading state.

I've made two tickets to extend off this:
[Use local loading](https://github.com/airbytehq/airbyte/issues/20536)
[Toast errors](https://github.com/airbytehq/airbyte/issues/20538)

**Review by dizel852 — APPROVED** [review]
LGTM

**Review by krishnaglick — APPROVED** [review]

**Review by teallarson — APPROVED** [review]
tested locally with flag off and on, creating and editing connections (did not have any connections from before this PR to test) 

code lgtm
